### ISSUE TITLE
GDAX API is down! unable to call getBalance

### ISSUE THREAD

**Issue by ColterD** [DeviaVir/zenbot#669]
Same issue as #519 

> GDAX API is down! unable to call getBalance, retrying in 10s
> { Error: non-200 status: 400
>     at statusErr (/home/main/zenbot/extensions/exchanges/gdax/exchange.js:30:17)
>     at /home/main/zenbot/extensions/exchanges/gdax/exchange.js:93:25
>     at Request._callback (/home/main/zenbot/node_modules/gdax/lib/clients/public.js:45:7)
>     at Request.self.callback (/home/main/zenbot/node_modules/gdax/node_modules/request/request.js:187:22)
>     at emitTwo (events.js:125:13)
>     at Request.emit (events.js:213:7)
>     at Request.<anonymous> (/home/main/zenbot/node_modules/gdax/node_modules/request/request.js:1044:10)
>     at emitOne (events.js:115:13)
>     at Request.emit (events.js:210:7)
>     at IncomingMessage.<anonymous> (/home/main/zenbot/node_modules/gdax/node_modules/request/request.js:965:12)
>     at emitNone (events.js:110:20)
>     at IncomingMessage.emit (events.js:207:7)
>     at endReadableNT (_stream_readable.js:1056:12)
>     at _combinedTickCallback (internal/process/next_tick.js:138:11)
>     at process._tickCallback (internal/process/next_tick.js:180:9)
>   code: 'HTTP_STATUS',
>   body: { message: 'request timestamp expired' } }
> [ { currency: 'USD', asset: 'BTC' } ]
> 

[GDax's API documentation](https://docs.gdax.com/?javascript#api) mentions that:
400 | Bad Request – Invalid request format
-but everything is set according to Zenbot documentation?

API keys are correct with right permissions.
Files are set with execute permissions (if that would matter in this case).
OS is Debian GNU/Linux 8.9 (jessie).
Server is located in Dallas, TX and API isn't going off of whitelisted IP.
Any ideas?

**Comment by ColterD** [comment]
Will be closing this issue, resolved.

People of the future:
I did some research on some basic API principles (since i'm clueless about API mostly), this happened because my local server time wasn't synced up with GDAX server time.  The way I fixed this was:

> sudo apt-get install ntp
> sudo systemctl restart ntp
> sudo systemctl enable ntp

Edit from a month later:   Also, you might want to check if your timezone is set to YOUR timezone.
> dpkg-reconfigure tzdata

My time was mostly up to date before, but it was more than 30 seconds off of GDAX's time so it would error out.  Run these 3 commands, check your **$ date** and it should be more in-sync than it was before.  Such a simple fix to a stupid problem.
